core/state/state_object.go

```diff
@@ -203,6 +206,8 @@ func (s *stateObject) GetCommittedState(key common.Hash) common.Hash {
 				s.db.setError(err)
 			}
 			value.SetBytes(content)
+		} else {
+			s.isFilled[key] = true // if we get here, it means that the fill wasn't cached and so it's a fill
```

Comment:
Any storage-slot write _must_ have called `GetCommitedState(...)` before for this map to be updated properly for a subsequent _further_ witness recording. If in any current (or future) case this doesn't hold, the gas will be accounted incorrectly.

Isn't this quite dangerous assumption?